src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -132,30 +135,158 @@
             (str "Unable to find bootstrap.cfg file via --bootstrap-config "
                  "command line argument, current working directory, or on classpath")))))
 
-(schema/defn chain-files :- [schema/Str]
-  "Takes a list of files, reads all their lines in, and returns a flattened seq
-   of all their lines put together.
+(schema/defn indexed
+  "Returns seq of [index, item] pairs
+  [:a :b :c] -> ([0 :a] [1 :b] [2 :c])"
+  [coll]
+  (map vector (range) coll))
+
+(schema/defn read-config :- [schema/Str]
+  "Opens a bootstrap file from either a path or URI, and returns each line from
+  the config. Throws an exception if the file can't be found or if the URI
+  can't be loaded"
+  [config-path :- schema/Str]
+  (let [config-file (if (fs/file? config-path)
+                      (fs/file config-path)
+                      (try
+                        (io/input-stream (URI. config-path))
+                        (catch Exception ignored
```

Comment:
Generally speaking it's Very Bad Juju to catch and swallow an exception type that is this generic.  Is there something more specific you could catch?

My guess is that if you pulled the `URI` constructor call out into a `let` block, you would be able to catch some kind of "malformed" exception for that, and then maybe catch an IOException for the call to `io/input-stream`?

If you head in that direction, would be good to add basic unit tests for `read-config` for both of those cases.